src/query/processor.rs

```diff
@@ -114,14 +116,11 @@ impl<T: Transport + Clone> Processor<T> {
         let handle = self.queries.handle(query_id);
         handle.set_state(QueryState::Preparing(req))?;
 
-        // invariant: this helper's identity must be the first element in the array.
-        let this = self.identities[0];
-        let right = self.identities[1];
-        let left = self.identities[2];
+        let id = self.identity();
+        let [right, left] = id.others();
```

Comment:
This relates to #376.

I added a test `helpers::tests::role_assignment_tests::multiply_with_various_roles` that is covering basic functionality in both ring orders, and it works.

I did try randomly assigning to `right` and `left` here, and it somewhat worked, but I didn't resolve all of the test failures. It's possible that the failures are just due to not being careful enough about which arrays are helper-indexed vs. role-indexed, and could be fixed easily (along the lines of the fix to `TestWorld::new_with` in this diff), but I wasn't sure if we really wanted to add random role assignments as a variable in our work right now. If there's interest in that, I'm happy to look at the rest of the test failures.

Reply:
yea it does not seem we can handle it. It would be great to have canonical order and stick with it for some time - for 2023 it would probably be more than enough